src/uaserver.cpp

```diff
@@ -101,8 +108,42 @@ void UaServer::setServerConfig(
         const UaString& configurationFile,
         const UaString& applicationPath)
 {
-    LOG(Log::INF) << "Note: with open62541 backend, there isn't (yet) XML configuration loading. Assuming hardcoded server settings (endpoint's port, etc.)";
+#ifndef HAS_SERVERCONFIG_LOADER
+    LOG(Log::INF) << "Note: you built open62541-compat without configuration loading (option SERVERCONFIG_LOADER). So loading of ServerConfig.xml is not supported. Assuming hardcoded server settings (endpoint's port, etc.)";
     //! With open62541 1.0, it is the UA_Server that holds the config.
+#else // HAS_SERVERCONFIG_LOADER is defined, means the user wants the option
+    std::unique_ptr< ::ServerConfig::OpcServerConfig > serverConfig;
+     try
+     {
+         serverConfig = ServerConfig::OpcServerConfig_ (configurationFile.toUtf8());
+     }
+     catch (xsd::cxx::tree::parsing<char> &exception)
+     {
+         LOG(Log::ERR) << "ServerConfig loader: failed when trying to open the file, with general error message: " << exception.what();
+         for( const xsd::cxx::tree::error<char> &error : exception.diagnostics() )
+         {
+             LOG(Log::ERR) << "ServerConfig: Problem at " << error.id() << ":" << error.line() << ": " << error.message();
+         }
+         throw std::runtime_error("ServerConfig: failed to load ServerConfig. The exact problem description should have been logged.");
+
+     }
+     // minimum one endpoint is guaranteed by the XSD, but in case user declared more, refuse to continue
```

Comment:
Is this required if the xsd defines it as a singular thing? I mean wouldn't line 118 throw an exception?

Reply:
It would, without any detailed information on what's wrong with your document (i.e. WHY it failed schema validation). In this fashion you also learn at which line there is the problem, and why.